### PR TITLE
[VC] Add simple first fit namespace scheduling algorithm

### PR DESCRIPTION
**Pull request by Fei-Guo**
This change adds a simple namespace scheduling algorithm and adds the missing parts in the end-to-end workflow.

### CODE REVIEW

File: incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/namespace/controller.go [hunk 1]
```diff
@@ -149,16 +156,36 @@ func (c *controller) Reconcile(request reconciler.Request) (reconciler.Result, e
...
+	// update virtualcluster namespace with the scheduling result.
+	vcClient, err := c.MultiClusterController.GetClusterClient(request.ClusterName)
 	if err != nil {
-		return reconciler.Result{Requeue: true}, fmt.Errorf("failed to schedule namespace %s in %s: %v", request.Name, request.ClusterName, err)
+		return reconciler.Result{}, fmt.Errorf("failed to get vc %s's client: %v", request.ClusterName, err)
+	}
+	updatedPlacement, _ := json.Marshal(ret.GetPlacementMap())
+	clone := namespace.DeepCopy()
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		clone.Annotations[utilconst.LabelScheduledPlacements] = string(updatedPlacement)
```
Comment (zhuangqh) [inline incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/namespace/controller.go:177]: annotation may be nil

**Review by zhuangqh — COMMENTED** [review]
minor issue

File: incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/namespace/controller.go (hunk 1 above)
Comment (Fei-Guo) [reply incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/namespace/controller.go:180]: fixed

**Comment by Fei-Guo** [comment]
All fixed. Thanks for the review.

**Pull request merged**